File: tests/fixtures/sample.ts
/**
 * Sample TypeScript file for parser tests.
 * Contains functions, classes, interfaces, and imports.
 */

import { readFileSync } from "fs";
import path from "path";

const CONSTANT = "hello";

/** Add two numbers and return the result. */
function standaloneFunction(x: number, y: number): number {
  return x + y;
}

function functionNoJsdoc(name: string): string {
  return `Hello, ${name}`;
}

interface SampleInterface {
  id: number;
  name: string;
  process(): void;
}

type SampleType = {
  value: string;
  count: number;
};

/** A sample class with multiple methods. */
class SampleClass implements SampleInterface {
  public id: number;
  public name: string;

  constructor(id: number, name: string) {
    this.id = id;
    this.name = name;
  }

  /** Execute the processing logic. */
  process(): void {
    console.log(`Processing ${this.name}`);
  }

  static create(id: number): SampleClass {
    return new SampleClass(id, "default");
  }
}

function callsOtherFunctions(): number | null {
  /** Demonstrates call edges to standaloneFunction. */
  const result = standaloneFunction(1, 2);
  const filePath = path.join(".", "sample.ts");
  const _content = readFileSync(filePath);
  return result > 0 ? result : null;
}

export { standaloneFunction, SampleClass, SampleInterface, SampleType };

File: tests/fixtures/sample.py
"""Sample Python file for parser tests.

This file intentionally contains a variety of Python constructs:
functions, classes, methods, imports, and docstrings.
"""

import os
from pathlib import Path

CONSTANT = "hello"


def standalone_function(x: int, y: int) -> int:
    """Add two integers and return the result."""
    return x + y


def function_no_docstring(name: str) -> str:
    return f"Hello, {name}"


class SampleClass:
    """A sample class with multiple methods."""

    class_var: str = "class_var"

    def __init__(self, value: int) -> None:
        self.value = value

    def instance_method(self) -> int:
        """Return the stored value."""
        return self.value

    @staticmethod
    def static_method() -> str:
        return "static"

    @classmethod
    def class_method(cls) -> "SampleClass":
        return cls(0)


def calls_other_functions() -> int | None:
    """Calls standalone_function to demonstrate call edges."""
    result = standalone_function(1, 2)
    path = Path(os.getcwd())
    return result if path.exists() else None
